connect/client/models/resourceset.py

```diff
@@ -463,9 +481,10 @@ def __getitem__(self, key):  # noqa: CCR001
 
         copy = self._copy()
         copy._offset = key.start
-        copy._limit = key.stop - key.start
-        copy._slice = True
-        return copy
+        copy._slice = key
+        if copy._slice.stop - copy._slice.start < copy._limit:
+            copy._limit = copy._slice.stop - copy._slice.start
+        raise NotYetEvaluatedError()
```

Comment:
It's a bit complicated for me to merge having fixed just sync client, can you do the same for async one ?

Reply:
Yes, it is in progress